tree/30_generic_methods/_file_check_special.cf

```diff
@@ -44,13 +49,17 @@ bundle agent file_check_special(file_name, file_type)
   methods:
     exists_special::
       "file_exists_and_is_special"
+        usebundle => _classes_success("${old_class_prefix}");
+      "file_exists_and_is_special"
         usebundle => _classes_success("${class_prefix}");
```

Comment:
We would normally only call the _classes_success method for the new class_prefix if the has_promiser_stack class is defined. See for example https://github.com/Normation/ncf/blob/master/tree/30_generic_methods/_service_check_running_src.cf#L50.